packages/cordova-builders/cordova-serve/index.ts

```diff
@@ -6,7 +6,7 @@ import type {
   DevServerBuilderOptions,
   DevServerBuilderOutput,
 } from '@angular-devkit/build-angular';
-import type { IndexHtmlTransform } from '@angular-devkit/build-angular/src/utils/index-file/index-html-generator';
+import type { IndexHtmlTransform } from '@angular/build/src/utils/index-file/index-html-generator';
```

Comment:
We should avoid using internal imports. This leads to fragile code that can break unexpectedly between patch and minor releases. 

The return type of this seems straightforward enough to replace in the implementation below with:
```ts
export const indexHtmlTransformFactory: (
  formattedAssets: FormattedAssets,
  context: BuilderContext
) => (content: string) => Promise<string> =
```

Reply:
The index-html-generator in angular-devkit is no longer there, but it seems to have been refactored to @angular/build. The previous implementation also used an internal link, doesn't it?

I have implemented your alternate code.

Reply:
You are correct, the previous implementations used internal links. I can only help reduce the occurrence of it moving forward 😄 this should lead to less changes needed between versions in the future, unless Angular actually changes the type signature.